src/qb4-packw/gen/qb4-packw-x16c4-gemm-goi-neondot.c

```diff
@@ -0,0 +1,405 @@
+          uint32x4_t w9x0123 = vld1q_u32((uint32_t*) w9);
+          uint32x4_t wAx0123 = vld1q_u32((uint32_t*) w10);
+          uint32x4_t wBx0123 = vld1q_u32((uint32_t*) w11);
+          uint32x4_t wCx0123 = vld1q_u32((uint32_t*) w12);
+          uint32x4_t wDx0123 = vld1q_u32((uint32_t*) w13);
+          uint32x4_t wEx0123 = vld1q_u32((uint32_t*) w14);
+          uint32x4_t wFx0123 = vld1q_u32((uint32_t*) w15);
+
+          uint32x4_t v01_02 = vtrn1q_u32(w0x0123, w1x0123);
+          uint32x4_t v01_13 = vtrn2q_u32(w0x0123, w1x0123);
+          uint32x4_t v23_02 = vtrn1q_u32(w2x0123, w3x0123);
+          uint32x4_t v23_13 = vtrn2q_u32(w2x0123, w3x0123);
+          uint32x4_t v45_02 = vtrn1q_u32(w4x0123, w5x0123);
+          uint32x4_t v45_13 = vtrn2q_u32(w4x0123, w5x0123);
+          uint32x4_t v67_02 = vtrn1q_u32(w6x0123, w7x0123);
+          uint32x4_t v67_13 = vtrn2q_u32(w6x0123, w7x0123);
+          uint32x4_t v89_02 = vtrn1q_u32(w8x0123, w9x0123);
+          uint32x4_t v89_13 = vtrn2q_u32(w8x0123, w9x0123);
+          uint32x4_t vAB_02 = vtrn1q_u32(wAx0123, wBx0123);
+          uint32x4_t vAB_13 = vtrn2q_u32(wAx0123, wBx0123);
+          uint32x4_t vCD_02 = vtrn1q_u32(wCx0123, wDx0123);
+          uint32x4_t vCD_13 = vtrn2q_u32(wCx0123, wDx0123);
+          uint32x4_t vEF_02 = vtrn1q_u32(wEx0123, wFx0123);
+          uint32x4_t vEF_13 = vtrn2q_u32(wEx0123, wFx0123);
+
+          uint32x4_t v0123_0 = vcombine_u32(vget_low_u32(v01_02), vget_low_u32(v23_02));
+          uint32x4_t v0123_2 = vcombine_u32(vget_high_u32(v01_02), vget_high_u32(v23_02));
+          uint32x4_t v0123_1 = vcombine_u32(vget_low_u32(v01_13), vget_low_u32(v23_13));
+          uint32x4_t v0123_3 = vcombine_u32(vget_high_u32(v01_13), vget_high_u32(v23_13));
+          uint32x4_t v4567_0 = vcombine_u32(vget_low_u32(v45_02), vget_low_u32(v67_02));
+          uint32x4_t v4567_2 = vcombine_u32(vget_high_u32(v45_02), vget_high_u32(v67_02));
+          uint32x4_t v4567_1 = vcombine_u32(vget_low_u32(v45_13), vget_low_u32(v67_13));
+          uint32x4_t v4567_3 = vcombine_u32(vget_high_u32(v45_13), vget_high_u32(v67_13));
+          uint32x4_t v89AB_0 = vcombine_u32(vget_low_u32(v89_02), vget_low_u32(vAB_02));
+          uint32x4_t v89AB_2 = vcombine_u32(vget_high_u32(v89_02), vget_high_u32(vAB_02));
+          uint32x4_t v89AB_1 = vcombine_u32(vget_low_u32(v89_13), vget_low_u32(vAB_13));
+          uint32x4_t v89AB_3 = vcombine_u32(vget_high_u32(v89_13), vget_high_u32(vAB_13));
+          uint32x4_t vCDEF_0 = vcombine_u32(vget_low_u32(vCD_02), vget_low_u32(vEF_02));
+          uint32x4_t vCDEF_2 = vcombine_u32(vget_high_u32(vCD_02), vget_high_u32(vEF_02));
+          uint32x4_t vCDEF_1 = vcombine_u32(vget_low_u32(vCD_13), vget_low_u32(vEF_13));
+          uint32x4_t vCDEF_3 = vcombine_u32(vget_high_u32(vCD_13), vget_high_u32(vEF_13));
+
+          v0123_0 = xnn_packed2planar(&ksum0123, v0123_0, vmask, veor_mask, neg_zp);
```

Comment:
nit `veor_mask` - why does it have to be an arg?

Reply:
not sure what the alternative is for this? Do you mean to do the or outside of packed2planar?